File: Human_written_code/Code_without_comments/CPP/Q37.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

int main() {
    int t;
    cin >> t;

    while (t--) {
        string s;
        cin >> s;

        vector<int> cnt(2, 0);
        for (char c : s) {
            cnt[c - '0']++;
        }

        for (int i = 0; i <= s.size(); i++) {
            if (i == s.size() || cnt[1 - (s[i] - '0')] == 0) {
                cout << s.size() - i << endl;
                break;
            }
            cnt[1 - (s[i] - '0')]--;
        }
    }

    return 0;
}

File: Human_written_code/Code_without_comments/Python/Q37.py
def main():
    t = int(input())
    for _ in range(t):
        s = input()
        
        cnt = [0, 0]
        for c in s:
            cnt[int(c)] += 1

        for i in range(len(s) + 1):
            if i == len(s) or cnt[1 - int(s[i])] == 0:
                print(len(s) - i)
                break
            cnt[1 - int(s[i])] -= 1

if __name__ == "__main__":
    main()
